### PR TITLE
postprocessing_ffmpeg_wrapper

### PR DESCRIPTION
**Pull request by leclem**
quality controls

cutting video at the right time work

examples

script_example.py

Better quality function

changes working

work working

commit

solving bugs and code reviews

sonarcloud corrections

test_quality_check added

removing prints

removing url

generating zoom video in case regeneration fails

bug lower

new webp format

### CODE REVIEW

**Comment by leclem** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Vikit-ai/sdk/165?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#165** <a href="https://app.graphite.dev/github/pr/Vikit-ai/sdk/165?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Vikit-ai/sdk/165?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#164** <a href="https://app.graphite.dev/github/pr/Vikit-ai/sdk/164?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#163** <a href="https://app.graphite.dev/github/pr/Vikit-ai/sdk/163?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#162** <a href="https://app.graphite.dev/github/pr/Vikit-ai/sdk/162?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#161** <a href="https://app.graphite.dev/github/pr/Vikit-ai/sdk/161?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#160** <a href="https://app.graphite.dev/github/pr/Vikit-ai/sdk/160?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#159** <a href="https://app.graphite.dev/github/pr/Vikit-ai/sdk/159?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#158** <a href="https://app.graphite.dev/github/pr/Vikit-ai/sdk/158?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: vikit/wrappers/ffmpeg_wrapper.py [hunk 1]
```diff
@@ -618,6 +630,29 @@ async def reencode_video(video_url, target_video_name=None):
     if not target_video_name:
         target_video_name = "reencoded_" + get_canonical_name(video_url) + ".mp4"
 
+    logger.debug("Re-encoding video " + video_url + " with name " + target_video_name)
```
Comment (jeffmac-aix) [inline vikit/wrappers/ffmpeg_wrapper.py:633]: this gets very verbose, consider using logger.trace

File: vikit/wrappers/ffmpeg_wrapper.py [hunk 2]
```diff
@@ -664,6 +699,74 @@ async def reencode_video(video_url, target_video_name=None):
 
     return target_video_name
 
+async def cut_video(video_url, start_time, end_time, target_duration=None, target_video_name=None):
+    """
+    Cuts the video starting at start_time and ending at end_time
+    Args:
+        video_url (str): The video url to cut
+        start_time (float): The begining time of the video
```
Comment (jeffmac-aix) [inline vikit/wrappers/ffmpeg_wrapper.py:707]: please indicate the unit and format (may be using an example)

File: vikit/wrappers/ffmpeg_wrapper.py [hunk 3]
```diff
@@ -706,6 +810,80 @@ async def get_first_frame_as_image_ffmpeg(media_url, target_path=None):
     return target_path
 
 
+async def create_zoom_video(image_url, target_duration=3, target_video_name=None, ):
+    """
+    Cuts the video starting at start_time and ending at end_time
```
Comment (jeffmac-aix) [inline vikit/wrappers/ffmpeg_wrapper.py:815]: same description as cut_video, what is the difference then?

Comment (jeffmac-aix) [reply vikit/wrappers/ffmpeg_wrapper.py:815]: oh I see, we start at 0 (video start) and end at target duration? If so please just fix the first comment line :)

**Review by jeffmac-aix — CHANGES_REQUESTED** [review]

**Review by jeffmac-aix — APPROVED** [review]
lgtm

**Pull request merged**